### PR TITLE
Batch render user/agent messages & content fragments

### PR DESCRIPTION
**Pull request by PopDaph**

### CODE REVIEW

File: front/lib/api/assistant/conversation.ts [hunk 1]
```diff
@@ -200,61 +203,79 @@ async function renderUserMessage(
...
-          lastName: user.lastName,
-          fullName: user.firstName + (user.lastName ? ` ${user.lastName}` : ""),
-          image: null,
-          workspaces: [],
+  return messages.map((message) => {
+    const messageMentions = mentions.filter((m) => m.messageId === message.id);
+    const messageUser =
+      users.find((u) => u.id === message.userMessage?.userId) || null;
+
+    if (!message.userMessage) {
```
Comment (spolu) [inline front/lib/api/assistant/conversation.ts:225]: (We can move this 3 lines above and remove the ? in message.userMessage?)

Comment (PopDaph) [reply front/lib/api/assistant/conversation.ts:227]: It makes typescript unhappy to remove it 😬

File: front/lib/api/assistant/conversation.ts [hunk 2]
```diff
@@ -200,126 +205,210 @@ async function renderUserMessage(
...
+              acc.push(agentId);
+            }
+            return acc;
+          },
+          []
+        );
+        const agents = (
+          await Promise.all(
+            agentConfigurationIds.map(async (agentConfigId) => {
+              return await getAgentConfiguration(auth, agentConfigId);
```
Comment (PopDaph) [inline front/lib/api/assistant/conversation.ts:309]: I took a shortcut here, WDYT? I think it's worth shipping with it and iterate if we want to optimize again.

File: front/lib/api/assistant/conversation.ts [hunk 3]
```diff
@@ -200,126 +205,210 @@ async function renderUserMessage(
...
+            if (agentId && !acc.includes(agentId)) {
+              acc.push(agentId);
+            }
+            return acc;
+          },
+          []
+        );
+        return await Promise.all(
+          agentRetrievalActionIds.map(async (agentRetrievalActionId) => {
+            return await renderRetrievalActionByModelId(agentRetrievalActionId);
```
Comment (PopDaph) [inline front/lib/api/assistant/conversation.ts:329]: Same kind of shortcut here!

File: front/lib/api/assistant/conversation.ts (hunk 1 above)
Comment (spolu) [reply front/lib/api/assistant/conversation.ts:227]: Yes didn't mean removing it just moving it in place of line 223 :+1:

Comment (spolu) [reply front/lib/api/assistant/conversation.ts:227]: (to remove the ? at current l.225)

Comment (PopDaph) [reply front/lib/api/assistant/conversation.ts:223]: done!

**Review by spolu — DISMISSED** [review]
LGTM

File: front/lib/models/assistant/conversation.ts [hunk 4]
```diff
@@ -498,6 +498,9 @@ Message.init(
         unique: true,
         fields: ["conversationId", "rank", "version"],
       },
+      {
+        fields: ["conversationId"],
```
Comment (spolu) [inline front/lib/models/assistant/conversation.ts:502]: Should be covered by the index above?

Comment (PopDaph) [reply front/lib/models/assistant/conversation.ts]: I removed it and instead I updated the query to fetch the reactions to load only the sId of messages, which is the only field we use.

**Review by spolu — APPROVED** [review]

**Pull request merged**